### PR TITLE
SW-5072 Accession check-in improvements

### PR DESCRIPTION
**Pull request by alexTerraformation**
- Remove check-in confirmation dialog box
- Add ability to check-in accessions directly from the check-in route
- Add ability to bulk check-in all accessions

Screenshot 2024-03-20 at 9 49 25 AM
Screenshot 2024-03-20 at 9 50 59 AM
Screenshot 2024-03-20 at 10 53 12 AM

### CODE REVIEW

**Review by tommylau523 — APPROVED** [review]

**Review by karthikbtf — COMMENTED** [review]
Looks great and a nice upgrade 👍 .
A few comments/suggestions.

File: src/scenes/CheckIn/index.tsx [hunk 1]
```diff
@@ -66,6 +77,45 @@ export default function CheckIn(): JSX.Element {
...
+      }
+    }
+  };
+
+  const checkInAllAccessions = async () => {
+    if (pendingAccessions) {
+      try {
+        setBusyState(true);
+        await Promise.all(
+          pendingAccessions?.map(async (accession) => {
```
[the remark this answers is not in the document]
Comment (alexTerraformation) [reply src/scenes/CheckIn/index.tsx:104]: Good catch!

File: src/scenes/CheckIn/index.tsx [hunk 2]
```diff
@@ -148,6 +217,15 @@ export default function CheckIn(): JSX.Element {
...
                           />
+                          {userCanEdit && (
+                            <Button
+                              onClick={() => checkInAccession(result as Accession)}
+                              id='checkInCollection'
+                              label={strings.CHECK_IN}
+                              priority='secondary'
+                              type='productive'
+                            />
+                          )}
```
[the remark this answers is not in the document]
Comment (karthikbtf) [reply src/scenes/CheckIn/index.tsx:228]: I was thinking something like:

```
const checkInButton = useCallback((accession: Accession) => (
  {userCanEdit && (
    <Button
      onClick={() => 
      id...
     ...
    />
  )}
));

...
   {checkInButton(result)}
```

This seems a bit convoluted now and maybe not worth the refactor.

**Review by karthikbtf — APPROVED** [review]
Looks good to me, thanks for the changes @alexTerraformation

**Pull request merged**